docs/gallery_conf.py

```diff
@@ -0,0 +1,20 @@
+import os
+import sys
+
+import plotly.io as pio
+pio.renderers.default = 'sphinx_gallery'
+
+from mkdocs_gallery.gen_gallery import DefaultResetArgv
+from mkdocs_gallery.sorting import FileNameSortKey
+
+min_reported_time = 0
+if 'SOURCE_DATE_EPOCH' in os.environ:
+    min_reported_time = sys.maxint if sys.version_info[0] == 2 else sys.maxsize
+
+# To be used as the "base" config,
+# this script is referenced in the mkdocs.yaml under `conf_script` option: docs/gallery_conf.py
+conf = {
+    'reset_argv': DefaultResetArgv(),
+    'min_reported_time': min_reported_time,
+    'within_subsection_order': FileNameSortKey
```

Comment:
can we get some explanation for what these do?

Reply:
the first reset to defaults, I don't know why it is needed, I copy-pasted the config from 

https://github.com/smarie/mkdocs-gallery/blob/main/docs/gallery_conf.py

clearly it is not commented enough, nor it is in the docs

https://smarie.github.io/mkdocs-gallery/#2-configure-mkdocs

I'll add the comments of each one, the important one is `FileNameSortKey`, I explain in the ccn-template how to change sorting method, we can customize it. 
We should discuss about how to sort things

Reply:
in particular I describe the parameters here
https://ccn-template--39.org.readthedocs.build/en/39/notes/03-documentation/#set-up-mkdocs-gallery

Reply:
this is the call to sorted. the `__call__` method of self.conf["within_subsection_order"] returns the order you need.


```python
if sort_files:
    listdir = sorted(listdir, key=self.conf["within_subsection_order"]())
```

more details https://github.com/flatironinstitute/ccn-template/issues/37

Reply:
I think we want to link to the [sphinx-gallery config page](https://sphinx-gallery.github.io/stable/configuration.html), which has a good explanation -- you said the mkdocs-gallery dev is trying to duplicate what they're doing, right?

Reply:
From the sphinx-gallery docs, it looks like [DefaultResetArgv](https://sphinx-gallery.github.io/stable/gen_modules/sphinx_gallery.gen_gallery.html#sphinx_gallery.gen_gallery.DefaultResetArgv) does nothing -- it returns an empty list (whereas if it were not set, it [passes nothing](https://sphinx-gallery.github.io/stable/configuration.html#passing-command-line-arguments-to-example-scripts)). Does behavior change when this is set? For me building locally, doesn't look like it.